core/src/program/mod.rs

```diff
@@ -66,6 +71,12 @@ impl Program {
     }
 }
 
+impl AsRef<CodeBlock> for Program {
+    fn as_ref(&self) -> &CodeBlock {
+        self.root()
+    }
+}
```

Comment:
I'm hesitant to have this as a program is more than just a code block and dereferncing programs as code block may lead to confusion later on.

Reply:
This isn't being used anywhere and is now safely removed. Even tho these `AsRef` implementations are meant to expose a specific attribute of some struct, it might be better indeed to enforce the user to call a request for a code block explicitly